dss_datamover/master_application.py

```diff
@@ -179,6 +180,27 @@ def start(self):
         self.logger.info("Performing {} operation".format(self.operation))
         self.load_prefix_index_data()  # Load prefix metadata.
         self.load_prefix_keys_for_resume_operation()  # Check if resume operation required.
+
+        # Validate S3 prefix before starting the workers, to allow graceful exit of application for bad prefix
+        # (Fix for MIN-1312)
+        inv_prefix = 0
+        if self.prefix:
+            if not validate_s3_prefix(self.logger, self.prefix):
+                self.logger.fatal("Bad prefix specified, exit application.")
+                self.stop_logging()
+                sys.exit("Invalid prefix. Shutting down DataMover application")
+            cluster_ip = self.prefix.split('/')[0]
+            for nfs_share in self.config_nfs[cluster_ip]:
```

Comment:
If the cluster_ip in prefix is invalid IP address, then you get an exception for trying to use that in self.config_nfs

Reply:
Can you move the whole code into validate_s3_prefix function so that it looks nice. You can pass necessary parameters like self.prefix, self.config_nfs etc as positional ones and do the whole validation inside that function.

Reply:
updated in latest commit.